pyproject.toml

```diff
@@ -0,0 +1,153 @@
+
+# Enable the hatch-odoo metadata hook to generate dependencies from addons manifests.
+[tool.hatch.metadata.hooks.odoo-addons-dependencies]
+# Enable the hatch-odoo build hook to locate the Odoo addons and package them in odoo/addons.
+# if not provided, "locals" modules (those in './') are not found and cannot be installed...
+[tool.hatch.build.hooks.odoo-addons-dirs]
+
+[tool.hatch-odoo]
+# dependencies that are not in the addons manifest
+dependencies = [
+    "python-ldap",
+    "odoo==14.0.*",
+    # indirect unrealeased dependencies
+    "odoo-addons-design-themes==14.0.*",
+    # Add some constraints to Odoo deps that should be in their setup.py
+    "Babel==2.9.1",
+    "chardet==3.0.4",
+    "decorator==4.3.0",
+    "docutils==0.14",
+    "ebaysdk==2.1.5",
+    "freezegun==0.3.15",
+    "gevent==20.9.0 ",
+    # gevent==21.8.0 ",
+    "greenlet==1.1.2",
+    "idna==2.6",
+    "Jinja2==2.11.2",
+    "libsass>=0.18.0",
+    "lxml==4.6.5",
+    "Mako==1.0.7",
+    "MarkupSafe==1.1.0",
+    "num2words==0.5.6",
+    "ofxparse==0.19",
+    # ofxparse==0.21",
+    "passlib==1.7.1",
+    "Pillow==8.1.1",
+    "polib==1.1.0",
+    "psutil==5.6.6",
+    "psycopg2==2.9.10",
+    "pydot==1.4.1",
+    "python-ldap==3.1.0",
+    "PyPDF2==1.26.0",
+    "pyserial==3.4",
+    "python-dateutil==2.7.3",
+    "pytz",
+    "pyusb==1.0.2",
+    "qrcode==6.1",
+    "reportlab==3.5.55",
+    "requests==2.21.0",
+    # requests==2.25.1; python_version > '3.9'  # (Jammy) versions < 2.25 aren't compatible w/ urllib3 1.26. Bullseye = 2.25.1. min version = 2.22.0 (Focal)",
+    # urllib3==1.26.5; python_version > '3.9'  # (Jammy) indirect / min version = 1.25.8 (Focal with security backports)",
+    "zeep==3.2.0",
+    "python-stdnum==1.8",
+    "vobject==0.9.6.1",
+    "Werkzeug==0.16.1 ",
+    # Werkzeug==2.0.2 ; python_version > '3.9'  # (Jammy)",
+    "XlsxWriter==1.1.2",
+    "xlwt==1.3.*",
+    "xlrd==1.2.0",
+    # TODO: add constraints in github connector module
+    "PyGithub<2.0.0",
+]
+addons_dirs = ["./"]
+odoo_version_override = "14.0"
+
+[tool.uv]
+# On OCA 14.0 series translations commits are realeased
+# in pre-release state
+prerelease = "allow"
+# tell uv when dependencies may change
+cache-keys = [
+    { file = "pyproject.toml" },
+    { file = "**/__manifest__.py" },
+]
+
+dev-dependencies = [
+  "parameterized>=0.9.0",
+  "pytest>8",
+  "pytest-odoo>=2",
+  "pytest-cov>=5.0.0",
+  "pytest-xdist>=3.6.1",
+  "manifestoo>=1.0",
+  "odoo-test-helper",
+]
+
+
+[tool.uv.sources]
+odoo = { git = "https://github.com/petrus-v/odoo", branch = "14.0-oca-20250317" }
+odoo-addons-design-themes = { git = "https://github.com/petrus-v/design-themes", branch = "14.0-oca-20250317" }
+# PR: https://github.com/OCA/vertical-association/pull/151
+odoo14-addon-membership-delegated-partner-line = { git = "https://github.com/gdgellatly/vertical-association", branch = "gdg-membership_delegated_partner_line", subdirectory = "setup/membership_delegated_partner_line"}
```

Comment:
```suggestion
odoo14-addon-membership-delegated-partner-line = { git = "https://github.com/OCA/vertical-association", ref = "refs/pull/151/head", subdirectory = "setup/membership_delegated_partner_line"}
```

Reply:
I've already test that, but `ref` doesn't seems to be supported by uv even on the current released moving uv from 0.5.11 to 0.68 version but doesn't seems any change (there is the new package field)

```
unknown field `ref`, expected one of `git`, `subdirectory`, `rev`, `tag`, `branch`, `url`, `path`, `editable`, `package`, `index`, `workspace`, `marker`, `extra`, `group`
```

but `rev` seems to work as expected \o/ !